kotlinpoet/src/jvmTest/kotlin/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -4416,6 +4416,47 @@ class TypeSpecTest {
 
     assertThat(toString(type)).isEqualTo(expect)
   }
+// https://github.com/square/kotlinpoet/issues/2033
+  @Test fun testDelegateOnAnonymousObject() {
+    val type = TypeSpec.anonymousClassBuilder()
+      .addSuperinterface(
+        Consumer::class.parameterizedBy(String::class),
+        CodeBlock.of("java.util.function.Consumer({ println(it) })"),
+      )
+      .build()
+
+    val expect = """
+        |object : java.util.function.Consumer<kotlin.String> by java.util.function.Consumer({ println(it) }) {
+        |}
+    """.trimMargin()
+
+    assertThat(type.toString()).isEqualTo(expect)
+  }
+  // https://github.com/square/kotlinpoet/issues/2033
+  @Test fun testMultipleDelegatesOnAnonymousObject() {
+    val type = TypeSpec.anonymousClassBuilder()
+      .primaryConstructor(
+        FunSpec.constructorBuilder()
+          .build(),
+      )
+      .addSuperinterface(
+        Function::class.parameterizedBy(String::class, Int::class),
+        CodeBlock.of("kotlin.Function ({ text -> text.toIntOrNull() ?: 0 })"),
+      )
+      .addSuperinterface(
+        Runnable::class,
+        CodeBlock.of("java.lang.Runnable ({ %T.debug(\"Hello world\") })", Logger::class.asTypeName()),
```

Comment:
nit - this should not be necessary:

```suggestion
        CodeBlock.of("java.lang.Runnable ({ %T.debug(\"Hello world\") })", Logger::class),
```

Reply:
you're correct. I'll also remove it from the orig test